packages/flutter/test/material/dropdown_form_field_test.dart

```diff
@@ -195,6 +195,95 @@ void main() {
     expect(value, equals('three'));
   });
 
+  testWidgets('DropdownButtonFormFieldState value updates when a different option is selected', (WidgetTester tester) async {
+    const String value = 'one';
+    final GlobalKey<FormFieldState<String>> stateKey = GlobalKey<FormFieldState<String>>();
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: DropdownButtonFormField<String>(
+            key: stateKey,
+            value: value,
+            onChanged: (String newValue) {},
+            items: menuItems.map((String value) {
+              return DropdownMenuItem<String>(
+                value: value,
+                child: Text(value),
+              );
+            }).toList(),
+          ),
+        ),
+      )
+    );
+
+    // [FormField] value is properly initialized.
+    expect(stateKey.currentState.value, equals('one'));
+
+    // Tap [DropdownButton] to open dropdown menu.
+    await tester.tap(find.text('one'));
+    await tester.pumpAndSettle();
+
+    // Tap [DropdownMenuItem] for an item other than the initial item.
+    await tester.tap(find.text('three').last);
+    await tester.pumpAndSettle();
+
+    // [FormField] should update to selected item value.
+    expect(stateKey.currentState.value, equals('three'));
+  });
+
+  testWidgets('DropdownButtonFormFieldState internal InputDecorator isEmpty property is correct', (WidgetTester tester) async {
```

Comment:
It’s better to also check for the visual changes of having ‘isEmpty’ true/false, similar to that test I pointed out. 

The issue with a test like this is while the InputDecorator might be providing the correct ‘isEmpty’ value, the intended visual properties might not be adopted and this test will not catch that.

Reply:
Would a `expect(find.text('Hint'), findsOneWidget);` followed by testing the reverse situation be sufficient?

Reply:
It's the check for the animation and label heights [seen in this test](https://github.com/flutter/flutter/blob/master/packages/flutter/test/material/input_decorator_test.dart#L163)

Reply:
I'm not sure I understand the interactions correctly, but wouldn't the existing test that you linked already check for that assuming the `isEmpty` value is correct, since `DropdownButtonFormField` uses a normal `InputDecorator`?

Reply:
The concern I have is in the situation where even though the parameter `isEmpty` returns the correct value, the actual widget decoration may not be properly applied because of some unusual interaction between `InputDecorator` and `DropdownButtonFormField`. 

I've seen tests in the past for widgets that use other widgets as descendants "pass" when there is actually a bug. Typically, these tests check for a property value, but since the property was not properly connected to its descendants or because there was an unintended interaction between the widgets, the effect wasn't properly applied, resulting in a bug.

That's why I prefer checking the visual properties of a widget with particular properties, since that is the most sure way of verifying that we are getting the result we want.

Reply:
I realize that the term `visual properties` is vague, maybe "the actual positioning on the screen" might be more apt? :)